feature-libs/product/configurators/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+        ],
+      })
+        .overrideComponent(ConfiguratorAttributeProductCardComponent, {
+          set: {
+            changeDetection: ChangeDetectionStrategy.Default,
+          },
+        })
+        .compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      ConfiguratorAttributeProductCardComponent
+    );
+    component = fixture.componentInstance;
+    component.quantity = mockQuantity;
+    component.disabledAction = false;
+    component.multiSelect = false;
+    component.product = createValue(
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+      [createImage('url', 'alt')],
+      1,
+      false,
+      '1111-2222',
+      'Lorem Ipsum Dolor'
+    );
+
+    spyOn(component, 'onHandleSelect').and.callThrough();
+    spyOn(component, 'onHandleDeselect').and.callThrough();
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should button be enabled when disabledAction = false & selected = false', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be enabled when disabledAction = false & selected = true', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be disabled when disabledAction = true & selected = true', () => {
+    component.disabledAction = true;
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.disabled).toBe(true);
+  });
+
+  it('should button be called with onHandleSelect', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    button.click();
+
+    fixture.detectChanges();
+
+    expect(component.onHandleSelect).toHaveBeenCalled();
+  });
+
+  it('should button be called with onHandleDeselect', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    button.click();
+
+    fixture.detectChanges();
+
+    expect(component.onHandleDeselect).toHaveBeenCalled();
+  });
+
+  it('should button be select when multiSelect = false & selected = false', () => {
```

Comment:
again with the name `.. multiSelect = false ..`, change to something not related with the logic

Reply:
renamed